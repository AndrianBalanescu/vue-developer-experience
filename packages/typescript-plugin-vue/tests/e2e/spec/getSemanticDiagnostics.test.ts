import Proto from 'typescript/lib/protocol';
import { createLanguageServerForTest } from '../server';
describe('getSemanticDiagnostics', () => {
  test('should report semantic issues in a vue file', async () => {
    const server = createLanguageServerForTest();
    server.openFile('fixture-semantic-diagnostics.vue');
    server.sendCommand(Proto.CommandTypes.SemanticDiagnosticsSync, <Proto.SemanticDiagnosticsSyncRequestArgs>{
      file: server.canonicalFileName('fixture-semantic-diagnostics.vue'),
    });
    await server.close();
    expect(server.responses).toHaveLength(1);
    const response: Proto.SemanticDiagnosticsSyncResponse = server.responses[0];
    expect(response.command).toBe(Proto.CommandTypes.SemanticDiagnosticsSync);
    expect(response.body).toHaveLength(1);
    expect((response.body as Proto.Diagnostic[]).map(item => [item.text, item.start])).toMatchSnapshot();
  });
  
  test.only('in template expression', async () => {
    const server = createLanguageServerForTest();
    server.openFile('template/semantic-diagnostics.vue');
    server.sendCommand(Proto.CommandTypes.SemanticDiagnosticsSync, <Proto.SemanticDiagnosticsSyncRequestArgs>{
      file: server.canonicalFileName('template/semantic-diagnostics.vue'),
    });
    await server.close();
    expect(server.responses).toHaveLength(1);
    const response: Proto.SemanticDiagnosticsSyncResponse = server.responses[0];
    expect(response.command).toBe(Proto.CommandTypes.SemanticDiagnosticsSync);
    expect(response.body).toHaveLength(1);
    expect((response.body as Proto.Diagnostic[]).map(item => [item.text, item.start])).toMatchSnapshot();
  });
});
