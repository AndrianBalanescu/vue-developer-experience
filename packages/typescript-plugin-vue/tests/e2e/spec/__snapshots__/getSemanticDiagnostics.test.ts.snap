// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSemanticDiagnostics should report semantic issues in a vue file 1`] = `
Array [
  Array [
    "Type '4' is not assignable to type 'string'.",
    Object {
      "line": 4,
      "offset": 7,
    },
  ],
]
`;
